### PR TITLE
[Dashboard] Add computed total values to FundsCards

### PR DESCRIPTION
**Pull request by mmioana**
Add computed total values to FundsCards
Add loading skeletons to fund card components
Add useSubDomains hook
Moved convertFromTokenToCurrency and getValuesTrend to their own files

## Description

- This PR makes use of the cron lambda developed in this PR [3080](https://github.com/JoinColony/colonyCDapp/pull/3080) for caching all previous domain-level and colony-level balances for both the previous last 30 days and the previous total amount from the start of last week.

Streaming widget

- This PR refactors the `FundsCards` to:
  - use `useGetDomainBalanceQuery` for the displayed value
  - use `useGetCachedDomainBalanceQuery` for the trend
  - adds the loading skeletons for the cards
  - filters the domains showing up in the row based on the selected domain

> [!IMPORTANT]
> Given we don't have any cached data for the previous values, we'll need to perform some manual queries. 

* Step 1. First things first, run your dev environment
* Step 2. Run the `create-data` script
* Step 3. For ease of testing (kind of) please save the following code in a new file (you can name it `.diff`)
```
diff --git a/amplify/backend/api/colonycdapp/schema/schema.graphql b/amplify/backend/api/colonycdapp/schema/schema.graphql
index fa83b314f..4b99b0358 100644
--- a/amplify/backend/api/colonycdapp/schema/schema.graphql
+++ b/amplify/backend/api/colonycdapp/schema/schema.graphql
@@ -3373,6 +3373,7 @@ type ColonyAction @model @searchable {
   Approved tokens impacted by the action (used for manage tokens)
   """
   approvedTokenChanges: ApprovedTokenChanges
+  updatedAt: AWSDateTime!
 }
 
 """
diff --git a/src/graphql/generated.ts b/src/graphql/generated.ts
index 9ff935173..326dc5333 100644
--- a/src/graphql/generated.ts
+++ b/src/graphql/generated.ts
@@ -1456,6 +1456,7 @@ export type CreateColonyActionInput = {
   toPotId?: InputMaybe<Scalars['Int']>;
   tokenAddress?: InputMaybe<Scalars['ID']>;
   type: ColonyActionType;
+  updatedAt?: InputMaybe<Scalars['AWSDateTime']>;
 };
 
 export type CreateColonyActionMetadataInput = {
@@ -2856,6 +2857,7 @@ export type ModelColonyActionConditionInput = {
   toPotId?: InputMaybe<ModelIntInput>;
   tokenAddress?: InputMaybe<ModelIdInput>;
   type?: InputMaybe<ModelColonyActionTypeInput>;
+  updatedAt?: InputMaybe<ModelStringInput>;
 };
 
 export type ModelColonyActionConnection = {
@@ -2901,6 +2903,7 @@ export type ModelColonyActionFilterInput = {
   toPotId?: InputMaybe<ModelIntInput>;
   tokenAddress?: InputMaybe<ModelIdInput>;
   type?: InputMaybe<ModelColonyActionTypeInput>;
+  updatedAt?: InputMaybe<ModelStringInput>;
 };
 
 export type ModelColonyActionMetadataConditionInput = {
@@ -4167,6 +4170,7 @@ export type ModelSubscriptionColonyActionFilterInput = {
   toPotId?: InputMaybe<ModelSubscriptionIntInput>;
   tokenAddress?: InputMaybe<ModelSubscriptionIdInput>;
   type?: InputMaybe<ModelSubscriptionStringInput>;
+  updatedAt?: InputMaybe<ModelSubscriptionStringInput>;
 };
 
 export type ModelSubscriptionColonyActionMetadataFilterInput = {
@@ -9358,6 +9362,7 @@ export type UpdateColonyActionInput = {
   toPotId?: InputMaybe<Scalars['Int']>;
   tokenAddress?: InputMaybe<Scalars['ID']>;
   type?: InputMaybe<ColonyActionType>;
+  updatedAt?: InputMaybe<Scalars['AWSDateTime']>;
 };
 
 export type UpdateColonyActionMetadataInput = {

```

The reason for this patch is to be able to manipulate the  `updatedAt` field for a colony action

* Step 4. Run `git apply .diff`
* Step 5. Now the interesting part - let's create some historical testing data 👨‍🦳 
Go to http://localhost:20002 and run the following mutation
```
mutation MyMutation {
  createColonyAction(
    input: {
      amount: "230000000000000000000" # any amount would be fine, though for this action to have a visible result use a value higher than 10^18
      showInActionsList: true
      initiatorAddress: "<LEELA_WALLET_ADDRESS_HERE>"
      blockNumber: 977
      colonyId: "<COLONY_ADDRESS_HERE>"
      fromDomainId: "<DOMAIN_ID_HERE>" # or just use <COLONY_ADDRESS_HERE_1>
      rootHash: "0x0000000000000000000000000000000000000000000000000000000000000000"
      toDomainId: "<ANOTHER_DOMAIN_ID_HERE>" # or just use <COLONY_ADDRESS_HERE_2>
      tokenAddress: "<TOKEN_ADDRESS_HERE>"
      type: MOVE_FUNDS
      updatedAt: "2024-09-09T14:55:07.825Z" # update this to any date previous to the start of last week
    }
  ) {
    amount
    blockNumber
    colonyId
    createdAt
    fromDomainId
    id
    recipientAddress
    rootHash
    toDomainId
    tokenAddress
    type
    updatedAt
  }
}
```
* Step 6. Run a few mutations in a couple of domains to get some testing data 
* Step 7. Now it's time to cache the values from the past so run the following query
```
query MyQuery {
  cacheAllDomainBalance {
    success
  }
}
```
* Step 8. You can test the cache has been created by running this query
```
query MyQuery {
  listCacheTotalBalances(
    filter: {domainId: {eq: "<DOMAIN_ID_HERE>"}}
  ) {
    items {
      colonyAddress
      date
      domainId
      id
      timeframePeriod
      timeframeType
      totalUSDC
      totalUSDCIn
      totalUSDCOut
    }
  }
}
```

* Step 9. Now please go to http://localhost:9091/planex and check the loading state for the `Total` cards row

Screenshot 2024-09-23 at 11 14 16

* Step 10. Check if the values and trends are showing up for the `Total` cards
Based on the mutations you previously ran (and also the domain for which you ran them), the first card should show the updated total value and the trend, while the remaining cards should show the updated total value

Screenshot 2024-09-23 at 10 59 34

Screenshot 2024-09-23 at 10 59 49

Screenshot 2024-09-23 at 10 59 41

* Step 11. Check if domain filtering is applied to this row of cards, as
- the `All teams` should show the `Total` card followed by the domains within the colony

Screenshot 2024-09-23 at 10 59 34

- the `General` should show the `Total General` card followed by the sub-domains

Screenshot 2024-09-23 at 10 59 41

- selecting any other domain should show the `Total <DOMAIN_NAME>` card followed by the `Create team` card

Screenshot 2024-09-23 at 10 59 49

Further testing: please be as creative as possible

## Diffs

**New stuff** ✨

* `FundsCardsItem` component 
* `useSubDomains` query
* `getValuesTrend` util function
* `convertFromTokenToCurrency` util function

**Changes** 🏗
* Make use of the `LoadingSkeleton` for all `FundsCards` partials

Contributes to #2851

### CODE REVIEW

**Review by Nortsova — COMMENTED** [review]
Hey, @mmioana! ✨ Thank you for this hard work! You are awesome 🤩 

I left small comments and then we can open for review! Thank you so much

**Review by rdig — APPROVED** [review]
Seems to be working fine. Nicely done.

Screenshot from 2024-09-24 11-43-05
Screenshot from 2024-09-24 11-43-22
Screenshot from 2024-09-24 11-44-28
Screenshot from 2024-09-24 11-44-46
Screenshot from 2024-09-24 11-44-58
Screenshot from 2024-09-24 11-45-05
Screenshot from 2024-09-24 11-45-13

File: src/components/v5/frame/ColonyHome/partials/FundsCards/FundsCards.tsx [hunk 1]
```diff
@@ -24,39 +22,21 @@ export const FundsCards = () => {
...
-                      value="15,779.00"
-                      prefix={currencySymbolMap[currency]}
-                    />
-                  }
-                />
-              }
-            />
-          );
-        })}
+        {subTeams?.map((item) => (
```
Comment (bassgeta) [inline src/components/v5/frame/ColonyHome/partials/FundsCards/FundsCards.tsx:32]: maybe we could first filter out the null items, amplify is pretty annoying regarding that, so maybe a `.filter(notMaybe)` before the map

File: src/components/v5/frame/ColonyHome/partials/FundsCards/partials/FundsCardsItem.tsx [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+import { useCurrencyContext } from '~context/CurrencyContext/CurrencyContext.ts';
+import Numeral from '~shared/Numeral/index.ts';
+import WidgetCards from '~v5/common/WidgetCards/index.ts';
+
+import { useTotalData } from '../hooks.ts';
+
+import { FundsCardsSubTitle } from './FundsCardsSubTitle.tsx';
+
+interface FundsCardsItemProps {
+  domainId?: string;
```
Comment (bassgeta) [inline src/components/v5/frame/ColonyHome/partials/FundsCards/partials/FundsCardsItem.tsx:14]: should this ever be `undefined`?

File: src/components/v5/frame/ColonyHome/partials/FundsCards/partials/FundsCardsTotalDescription.tsx [hunk 3]
```diff
@@ -1,25 +1,54 @@
...
+            decimals={nativeToken.decimals}
+            suffix={` ${nativeToken.symbol}`}
+          />
+        </span>
+      </LoadingSkeleton>
+      <LoadingSkeleton isLoading={isLoading} className="h-4 w-[30px] rounded">
+        <span className="flex items-center gap-1 font-medium text-blue-400">
+          <Arrow weight="bold" />
+
+          <span>{percent} Week</span>
```
Comment (bassgeta) [inline src/components/v5/frame/ColonyHome/partials/FundsCards/partials/FundsCardsTotalDescription.tsx:49]: we need i18n for the `Week` part I guess

**Review by bassgeta — APPROVED** [review]
Honestly the amount of work put into this is amazing, really good job evolving the lambda into something that works nicely!
I've left 3 comments but tbh they're not that important to solve now since they're QoL type of stuff
Data got cached:
image
Then it's created for all the domains I've used (4 of em total)
image
Without any filter:
image
Serenity
image
Andromeda:
image

image
I think I may have broken something when running the initial queries and the `General` domain has no income this last week causing this negative percentage into the 7th circle of hell? :D @mmioana just letting you know of this one if it's me or something regarding the UI calculations, this is my lambda return value
image

File: src/components/v5/frame/ColonyHome/partials/FundsCards/partials/FundsCardsItem.tsx (hunk 2 above)
Comment (bassgeta) [reply src/components/v5/frame/ColonyHome/partials/FundsCards/partials/FundsCardsItem.tsx:14]: yes for the all teams one

**Comment by mmioana** [comment]
You were right @bassgeta! 💯  there was an issue with the trend not being divided to the absolute value of the previous and then there was a need for getting the absolute trend value before formatting `const valueToFormat = isValueOverLimit ? upperLimit : trend.abs();`
With the last commit, it should be fixed 🙌

**Comment by mmioana** [comment]
Merging to avoid any rebase conflicts ✅

**Pull request merged**